cloudinit/sources/DataSourceEc2.py

```diff
@@ -193,9 +192,19 @@ def get_instance_id(self):
             return self.metadata['instance-id']
 
     def _maybe_fetch_api_token(self, mdurls, timeout=None, max_wait=None):
+        """ Get an API token for EC2 Instance Metadata Service.
+
+        On EC2. IMDS will always answer an API token, unless
+        the instance owner has disabled the IMDS HTTP endpoint or
+        the network topology conflics with the configured hop-limit.
+        """
         if self.cloud_name != CloudNames.AWS:
             return
 
+        url_params = self.get_url_params()
+        if url_params.max_wait_seconds <= 0:
+            return False
```

Comment:
I admit to copying this config validation from other calls to ```wait_for_urls```. I took it to mean I should not assume this the value of ```max_wait_seconds``` to be valid.

Reply:
In the case of Ec2, from docs,  it's actually how many seconds to wait  on a given url check before considering it as a non-viable source for metadata. In your case I don't think you want to skip the api token check if set to 0 because in that scenario people tried to reduce that timeout to make things faster, but on ec2 proper that would just break things fast without obtaining the token.

https://cloudinit.readthedocs.io/en/latest/topics/datasources/ec2.html#configuration

I think we should drop this part.

Reply:
done